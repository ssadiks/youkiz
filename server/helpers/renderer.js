import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import serialize from 'serialize-javascript';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Routes from '../../src/Routes';
import HeaderContainer from '../../src/components/container/HeaderContainer/HeaderContainer';

const renderer = (req, store) => {
  const muiTheme = getMuiTheme({
    userAgent: req.headers['user-agent']
  });

  const content = renderToString(
    <Provider store={store}>
      <MuiThemeProvider muiTheme={muiTheme}>
        <StaticRouter location={req.path} context={{}}>
          <div>
            {renderRoutes(Routes)}
          </div>
        </StaticRouter>
      </MuiThemeProvider>
    </Provider>
  );

  return `
    <html>
      <head>
        <link href="css/style.css" rel="stylesheet">
      </head>
      <body>
        <div id="root">${content}</div>
        <script>
          window.INITIAL_STATE = ${serialize(store.getState())}
        </script>
        <script src="js/vendors.js"></script>
        <script src="js/bundle.js"></script>
      </body>
    </html>
  `;
};

export default renderer;
