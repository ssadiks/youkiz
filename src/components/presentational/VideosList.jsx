import React, { Component } from 'react';
import {
  Link
} from 'react-router-dom'

class VideosList extends Component {

  componentWillMount() {
    this.props.fetchVideosAction();
  }

  componentWillReceiveProps(nextProps) {
    // console.log('rp', this.props.videosList);
    if(nextProps.videosList !== this.props.videosList) {
      // this.props.fetchVideosAction();
    }
  }

  render() {
    const { videosList } = this.props;

    return (
      <div>
        <h1>List of videos</h1>
        {
          videosList &&
          <div>
            {
              videosList.map(video => {
                const imageUrl = `https://i.ytimg.com/vi/${video.videoId}/mqdefault.jpg`;

                return (
                <div key={video.videoId}>
                  <Link to={`videos/${video._id}`}>
                    <img className="media-object" src={imageUrl} alt="" />
                    <span>{video.song}</span>
                    <span>{video.type}</span>
                  </Link>
                </div>
                )
              })
            }
          </div>
        }

      </div>
    );
  }
}

export default VideosList;