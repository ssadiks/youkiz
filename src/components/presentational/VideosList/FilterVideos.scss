.filterVideos {

  &__form {
    display: flex;
  }
}
