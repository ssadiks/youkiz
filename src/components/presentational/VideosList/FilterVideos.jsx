import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import AutoComplete from 'material-ui/AutoComplete';
import { DANCE_STYLE, DANCERS } from '../../../constants';
import { getArrayOfValue, filterArrayBy } from '../../../helpers';

class FilterVideos extends Component {
  constructor(props) {
    super(props);
    this.state = {
      typeDance: null
    };
  }

  onSubmitSearch = (e) => {
    e.preventDefault();

    const params = {
      filters: {
        dancers: [],
        type: ''
      },
      limit: 10
    };

    if (this.state.typeDance) {
      params.filters.type = DANCE_STYLE[this.state.value];
    }

    if (this.state.dancerMale) {
      params.filters.dancers.push(this.state.dancerMale);
    }

    if (this.state.dancerFemale) {
      params.filters.dancers.push(this.state.dancerFemale);
    }

    this.props.onSubmit(params);
  }

  handleChange = (event, index, typeDance) => this.setState({ typeDance });

  handleChangeDancersMale = dancerMale => this.setState({ dancerMale });
  handleChangeDancersFemale = dancerFemale => this.setState({ dancerFemale });

  render() {
    const dancersFemale = getArrayOfValue(filterArrayBy(DANCERS, 'sex', 'FEMALE'), 'dancer');
    const dancersMale = getArrayOfValue(filterArrayBy(DANCERS, 'sex', 'MALE'), 'dancer');

    return (
      <div className="filterVideos">
        <h1>Filter</h1>
        <form className="filterVideos__form" onSubmit={this.onSubmitSearch}>
          <SelectField
            floatingLabelText="Dance Style"
            value={this.state.typeDance}
            onChange={this.handleChange}
          >
            {
              DANCE_STYLE.map((DanceStyle, index) => (
                <MenuItem key={index} value={index} primaryText={DanceStyle} />
              ))
            }
          </SelectField>
          <AutoComplete
            floatingLabelText="Male Dancers"
            filter={AutoComplete.fuzzyFilter}
            dataSource={dancersMale}
            maxSearchResults={5}
            onNewRequest={this.handleChangeDancersMale}
            openOnFocus
          />
          <AutoComplete
            floatingLabelText="Female Dancers"
            filter={AutoComplete.fuzzyFilter}
            dataSource={dancersFemale}
            maxSearchResults={5}
            onNewRequest={this.handleChangeDancersFemale}
            openOnFocus
          />
          <button type="submit">Filter</button>
        </form>

      </div>
    );
  }
}

FilterVideos.propTypes = {
  onSubmit: PropTypes.func.isRequired
};

export default FilterVideos;
