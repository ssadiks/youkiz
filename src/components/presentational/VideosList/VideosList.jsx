import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Loader from '../Loader/Loader';
import FilterVideos from '../FilterVideo/FilterVideos';
import ListOfVideos from '../ListOfVideos/ListOfVideos';

class VideosList extends Component {
  componentWillMount() {
    this.props.fetchVideosAction();
  }

  componentWillReceiveProps(nextProps) {
    // console.log('rp', this.props.videosList);
    if (nextProps.videosList !== this.props.videosList) {
      // this.props.fetchVideosAction();
    }
  }

  handleDeleteVideo = (videoId) => {
    this.props.openModalDialog({
      title: 'Delete Video',
      message: 'Are you sure',
      onValid: () => {
        this.props.deleteVideoAction(videoId);
      },
      onCancel: () => this.props.hideModalDialog()
    });
  }

  render() {
    const { videosList, dancersList, deleteVideoAction, editVideo,
      userConnected } = this.props;

    return (
      <div className="o-container">
        {
          <FilterVideos
            onSubmit={params => this.props.fetchVideosAction(params)}
            fetchDancersAction={this.props.fetchDancersAction}
            dancersList={dancersList}
          />
        }
        {
          this.props.isPending ?
            <Loader /> :
            <ListOfVideos
              videosList={videosList}
              deleteVideoAction={deleteVideoAction}
              editVideo={editVideo}
              userConnected={userConnected}
              handleDeleteVideo={this.handleDeleteVideo}
            />
        }
      </div>
    );
  }
}

VideosList.defaultProps = {
  videosList: [],
  dancersList: [],
  userConnected: false
};

VideosList.propTypes = {
  videosList: PropTypes.array,
  dancersList: PropTypes.array,
  fetchVideosAction: PropTypes.func.isRequired,
  fetchDancersAction: PropTypes.func.isRequired,
  deleteVideoAction: PropTypes.func.isRequired,
  openModalDialog: PropTypes.func.isRequired,
  hideModalDialog: PropTypes.func.isRequired,
  editVideo: PropTypes.func.isRequired,
  isPending: PropTypes.bool.isRequired,
  userConnected: PropTypes.bool,
};

export default VideosList;
