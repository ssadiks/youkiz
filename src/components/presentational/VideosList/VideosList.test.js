import React from 'react';
import renderer from 'react-test-renderer';
import VideosList from './VideosList';

jest.mock('material-ui/IconMenu', () => 'IconMenu');
jest.mock('material-ui/MenuItem', () => 'MenuItem');
jest.mock('material-ui/IconButton', () => 'IconButton');

jest.mock('material-ui');

describe('VideosList render', () => {
  const videosList = [{ _id: 1, title: 'video 1' }, { _id: 2, title: 'video 2' }];
  const dancersList = [{ _id: 1, name: 'name 1' }, { _id: 2, name: 'name 2' }];

  it('if user is authentificated', () => {
    const tree = renderer.create(
      <VideosList
        videosList={videosList}
        dancersList={dancersList}
        userConnected
        editVideo={() => {}}
        handleDeleteVideo={() => {}}
        fetchVideosAction={() => {}}
        fetchDancersAction={() => {}}
        deleteVideoAction={() => {}}
        openModalDialog={() => {}}
        hideModalDialog={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('if user is not authentificated', () => {
    const tree = renderer.create(
      <VideosList
        videosList={videosList}
        dancersList={dancersList}
        userConnected={false}
        editVideo={() => {}}
        handleDeleteVideo={() => {}}
        fetchVideosAction={() => {}}
        fetchDancersAction={() => {}}
        deleteVideoAction={() => {}}
        openModalDialog={() => {}}
        hideModalDialog={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
