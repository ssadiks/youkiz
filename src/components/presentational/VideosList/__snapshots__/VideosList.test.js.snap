// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideosList render if user is authentificated 1`] = `
<div
  className="o-container"
>
  <div
    className="FilterVideos"
  >
    <h2>
      Filter
    </h2>
    <form
      className="FilterVideos__form"
      onSubmit={[Function]}
    >
      <div
        className="Select FilterVideos__select FilterVideos__select--typeDance Select--single is-clearable is-searchable"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-2--value"
          >
            <div
              className="Select-placeholder"
            >
              Select a dance style
            </div>
            <div
              className="Select-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <style
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "input#undefined::-ms-clear {display: none;}",
                  }
                }
              />
              <input
                aria-activedescendant="react-select-2--value"
                aria-describedby={undefined}
                aria-expanded="false"
                aria-haspopup="false"
                aria-label={undefined}
                aria-labelledby={undefined}
                aria-owns=""
                className={undefined}
                id={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                role="combobox"
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5px",
                  }
                }
                tabIndex={undefined}
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
      <div
        className="Select FilterVideos__select FilterVideos__select--dancers Select--multi is-clearable is-searchable"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-3--value"
          >
            <div
              className="Select-placeholder"
            >
              Select dancers
            </div>
            <div
              className="Select-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <style
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "input#undefined::-ms-clear {display: none;}",
                  }
                }
              />
              <input
                aria-activedescendant="react-select-3--value"
                aria-describedby={undefined}
                aria-expanded="false"
                aria-haspopup="false"
                aria-label={undefined}
                aria-labelledby={undefined}
                aria-owns=""
                className={undefined}
                id={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                role="combobox"
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5px",
                  }
                }
                tabIndex={undefined}
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
      <div
        className="FilterVideos__buttons"
      />
    </form>
  </div>
  <div
    className="ListOfVideos"
  >
    <h2>
      2 result(s)
    </h2>
    <div
      className="ListOfVideos__list"
    >
      <div
        className="ListOfVideos__item"
      >
        <a
          className="ListOfVideos__item__link"
          href="//www.youtube.com/watch?v=undefined"
          target="_blank"
        >
          <img
            alt=""
            className="media-object"
            src="https://i.ytimg.com/vi/undefined/mqdefault.jpg"
          />
        </a>
        <div
          className="ListOfVideos__item__description"
        >
          <span />
          <span />
          <IconMenu
            anchorOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            className="ListOfVideos__item__more"
            iconButtonElement={
              <IconButton>
                <NavigationMoreVert />
              </IconButton>
            }
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
          >
            <MenuItem
              onClick={[Function]}
              primaryText="Edit"
            />
            <MenuItem
              onClick={[Function]}
              primaryText="Delete"
            />
            <MenuItem
              primaryText="Infos"
            />
          </IconMenu>
        </div>
      </div>
      <div
        className="ListOfVideos__item"
      >
        <a
          className="ListOfVideos__item__link"
          href="//www.youtube.com/watch?v=undefined"
          target="_blank"
        >
          <img
            alt=""
            className="media-object"
            src="https://i.ytimg.com/vi/undefined/mqdefault.jpg"
          />
        </a>
        <div
          className="ListOfVideos__item__description"
        >
          <span />
          <span />
          <IconMenu
            anchorOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            className="ListOfVideos__item__more"
            iconButtonElement={
              <IconButton>
                <NavigationMoreVert />
              </IconButton>
            }
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
          >
            <MenuItem
              onClick={[Function]}
              primaryText="Edit"
            />
            <MenuItem
              onClick={[Function]}
              primaryText="Delete"
            />
            <MenuItem
              primaryText="Infos"
            />
          </IconMenu>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`VideosList render if user is not authentificated 1`] = `
<div
  className="o-container"
>
  <div
    className="FilterVideos"
  >
    <h2>
      Filter
    </h2>
    <form
      className="FilterVideos__form"
      onSubmit={[Function]}
    >
      <div
        className="Select FilterVideos__select FilterVideos__select--typeDance Select--single is-clearable is-searchable"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-4--value"
          >
            <div
              className="Select-placeholder"
            >
              Select a dance style
            </div>
            <div
              className="Select-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <style
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "input#undefined::-ms-clear {display: none;}",
                  }
                }
              />
              <input
                aria-activedescendant="react-select-4--value"
                aria-describedby={undefined}
                aria-expanded="false"
                aria-haspopup="false"
                aria-label={undefined}
                aria-labelledby={undefined}
                aria-owns=""
                className={undefined}
                id={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                role="combobox"
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5px",
                  }
                }
                tabIndex={undefined}
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
      <div
        className="Select FilterVideos__select FilterVideos__select--dancers Select--multi is-clearable is-searchable"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-5--value"
          >
            <div
              className="Select-placeholder"
            >
              Select dancers
            </div>
            <div
              className="Select-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <style
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "input#undefined::-ms-clear {display: none;}",
                  }
                }
              />
              <input
                aria-activedescendant="react-select-5--value"
                aria-describedby={undefined}
                aria-expanded="false"
                aria-haspopup="false"
                aria-label={undefined}
                aria-labelledby={undefined}
                aria-owns=""
                className={undefined}
                id={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                role="combobox"
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5px",
                  }
                }
                tabIndex={undefined}
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
      <div
        className="FilterVideos__buttons"
      />
    </form>
  </div>
  <div
    className="ListOfVideos"
  >
    <h2>
      2 result(s)
    </h2>
    <div
      className="ListOfVideos__list"
    >
      <div
        className="ListOfVideos__item"
      >
        <a
          className="ListOfVideos__item__link"
          href="//www.youtube.com/watch?v=undefined"
          target="_blank"
        >
          <img
            alt=""
            className="media-object"
            src="https://i.ytimg.com/vi/undefined/mqdefault.jpg"
          />
        </a>
        <div
          className="ListOfVideos__item__description"
        >
          <span />
          <span />
        </div>
      </div>
      <div
        className="ListOfVideos__item"
      >
        <a
          className="ListOfVideos__item__link"
          href="//www.youtube.com/watch?v=undefined"
          target="_blank"
        >
          <img
            alt=""
            className="media-object"
            src="https://i.ytimg.com/vi/undefined/mqdefault.jpg"
          />
        </a>
        <div
          className="ListOfVideos__item__description"
        >
          <span />
          <span />
        </div>
      </div>
    </div>
  </div>
</div>
`;
